Assignment/PingProcess.cs

```diff
@@ -13,54 +13,93 @@ public record struct PingResult(int ExitCode, string? StdOutput);
 
 public class PingProcess
 {
+    // used to decide whether we should use Windows or *nix ping flags
+    public static bool IsWindows => RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+
     private ProcessStartInfo StartInfo { get; } = new("ping");
 
+    private const int DefaultPingCount = 4;
+    private const int DefaultLongRunningPingCount = 8;
+
+    public static string CrossPlatformNumPingFlags(int pings) =>
+        IsWindows?$"-n {pings}":$"-c {pings}";
+
     public PingResult Run(string hostNameOrAddress)
     {
-        StartInfo.Arguments = hostNameOrAddress;
+        ArgumentException.ThrowIfNullOrEmpty(hostNameOrAddress);
+        string pingCountArg = CrossPlatformNumPingFlags(DefaultPingCount);
+        StartInfo.Arguments = $"{pingCountArg} {hostNameOrAddress}";
         StringBuilder? stringBuilder = null;
         void updateStdOutput(string? line) =>
             (stringBuilder??=new StringBuilder()).AppendLine(line);
         Process process = RunProcessInternal(StartInfo, updateStdOutput, default, default);
         return new PingResult( process.ExitCode, stringBuilder?.ToString());
     }
 
+    private Task<PingResult> BuildPingTask(string hostNameOrAddress, int pingCount = DefaultPingCount,
+        CancellationToken cancellationToken = default)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(hostNameOrAddress);
+        string pingCountArg = CrossPlatformNumPingFlags(pingCount);
+        StartInfo.Arguments = $"{pingCountArg} {hostNameOrAddress}";
+        StringBuilder? stringBuilder = null;
+        void updateStdOutput(string? line) =>
+            (stringBuilder ??= new StringBuilder()).AppendLine(line);
+
+        Task<PingResult> task = Task.Run(() => RunProcessInternal(StartInfo, updateStdOutput, default, cancellationToken))
+            .ContinueWith(t => new PingResult(t.Result.ExitCode, stringBuilder?.ToString()));
+
+        return task;
+    }
+
     public Task<PingResult> RunTaskAsync(string hostNameOrAddress)
     {
-        throw new NotImplementedException();
+        Task<PingResult> task = BuildPingTask(hostNameOrAddress);
+        task.Wait();
+        return task;

```

Comment:
Why wait? Why not just return the task?
```suggestion
        return BuildPingTask(hostNameOrAddress);
```